charts/opentelemetry-collector/templates/_helpers.tpl

```diff
@@ -153,13 +153,12 @@ Return if ingress is stable.
 {{- end }}
 
 {{- define "opentelemetry-collector.annotations" -}}
-{{- if eq .Values.mode "deployment" }}
-{{- if .Values.annotations }}
+{{- if or .Values.annotations .Values.standaloneCollector.annotations -}}
 annotations:
+{{- if .Values.annotations }}
 {{- .Values.annotations | toYaml | nindent 2  }}
-{{- end }}
-{{- else if .Values.standaloneCollector.annotations }}
-annotations:
+{{- else }}
```

Comment:
As per conversation when `.annotations` is set `.standaloneCollector.annotations` is ignored

Reply:
This version will still allow `.Values.standaloneCollector.annotations` values to be used when `mode` is set but `.Values.annotations` is not.  It's an unlikely scenario but technically possible.

I know this pain of supporting backwards compatibility with agentCollector and standaloneCollector lol if you're willing to wait a couple days, I will be removing agentCollector and standaloneCollector support as soon as collector 0.52.0 releases this week, which will make this this feature add simpler.